src/Renderer/MaterialLayer.js

```diff
@@ -82,8 +82,9 @@ class MaterialLayer {
         defineLayerProperty(this, 'zmin', layer.zmin, defaultEle.zmin);
         defineLayerProperty(this, 'zmax', layer.zmax, defaultEle.zmax);
 
-        this.textures = [];
-        this.offsetScales = [];
+        this.textures = new Map();
+        extents.forEach(ex => this.textures.set(ex, null));
+        this.offsetScales = new Map();
```

Comment:
You don't set `this.offsetScales` ?

Reply:
No, the previous `offsetScales` wasn't initializing.

Reply:
I thinks remove this property.

Reply:
done